secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/JwsSigner.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.signer;
+
+import java.util.Map;
+
+import org.forgerock.util.promise.Promise;
+
+/**
+ * Abstraction of the signature methods to implement a code segment to sign messages.
+ * <br/>
+ * <h3>Signer</h3>
+ * <p>
+ * A signer is a functional unity that implements this interface to sign messages. <br/>
+ * A signer implementation can be configured as IG heap object, where IG will create and initialise it in the heap environment
+ * to be used by the filters and/or the IG scripts, for more information see <a href='https://backstage.forgerock.com/docs/ig/7.2/reference/RequiredConfiguration.html#heap-objects'>IG spec reference</a><br/>
+ * <br/>
+ * <p>
+ * See:
+ * <ul>
+ *     <li>Signer implementation class: <em>com.forgerock.sapi.gateway.jws.signer.CompactSerializationJwsSigner</em></li>
+ *     <li>Configuration descriptor: <em>config.json</em></li>
+ *     <li>Route: <em>68-ob-aggregated-polling.json</em></li>
```

Comment:
Definitely don't mention all the places this is used! It is a library, it could be used from anywhere.

Reply:
deleted